// @stryker-mutator/core/dist/src/initializer/custom-initializers/svelte-initializer.js
import inquirer from 'inquirer';
const guideUrl = 'https://stryker-mutator.io/docs/stryker-js/guides/svelte';
const reporters = Object.freeze(['progress', 'clear-text', 'html']);
export class SvelteInitializer {
    name = 'svelte';
    async createConfig() {
        const testRunnerChoices = ['jest', 'vitest'];
        const testRunnerNodeArgs = [];
        const { testRunner } = await inquirer.prompt({
            choices: testRunnerChoices,
            message: 'Which test runner are you using?',
            name: 'testRunner',
            type: 'list',
        });
        if (testRunner === 'jest') {
            const { nativeEsm } = await inquirer.prompt({
                type: 'confirm',
                name: 'nativeEsm',
                message: 'Are you using native EcmaScript modules? (see https://jestjs.io/docs/ecmascript-modules)',
                default: true,
            });
            if (nativeEsm) {
                testRunnerNodeArgs.push('--experimental-vm-modules');
            }
        }
        return {
            config: {
                testRunner,
                ...(testRunnerNodeArgs.length ? { testRunnerNodeArgs } : {}),
                reporters,
            },
            dependencies: [`@stryker-mutator/${testRunner}-runner`],
            guideUrl,
        };
    }
}
